kmip/pie/client.py

```diff
@@ -0,0 +1,340 @@
+
+        # Verify that operations can be given at this time
+        if not self._is_open:
+            raise ClientConnectionNotOpen()
+
+        # Create the algorithm, length, and usage mask attributes
+        algorithm_attribute = self.attribute_factory.create_attribute(
+            AttributeType.CRYPTOGRAPHIC_ALGORITHM,
+            algorithm)
+        length_attribute = self.attribute_factory.create_attribute(
+            AttributeType.CRYPTOGRAPHIC_LENGTH,
+            length)
+        mask_attribute = self.attribute_factory.create_attribute(
+            AttributeType.CRYPTOGRAPHIC_USAGE_MASK,
+            [CryptographicUsageMask.ENCRYPT, CryptographicUsageMask.DECRYPT])
+
+        # Create the template containing the attributes
+        attributes = [algorithm_attribute, length_attribute, mask_attribute]
+        template = TemplateAttribute(attributes=attributes)
+
+        # Create the symmetric key and handle the results
+        result = self.proxy.create(ObjectType.SYMMETRIC_KEY, template)
+
+        status = result.result_status.enum
+        if status == ResultStatus.SUCCESS:
+            uid = result.uuid.value
+            return uid
+        else:
+            reason = result.result_reason.enum
+            message = result.result_message.value
+            raise KmipOperationFailure(status, reason, message)
+
+    def create_asymmetric_key_pair(self, algorithm, length):
+        """
+        Create an asymmetric key pair on a KMIP appliance.
+
+        Args:
+            algorithm (CryptographicAlgorithm): An enumeration defining the
+                algorithm to use to generate the key pair.
+            length (int): The length in bits for the key pair.
+
+        Raises:
+            ClientConnectionNotOpen: if the client connection is unusable
+            KmipOperationFailure: if the operation result is a failure
+            TypeError: if the input arguments are invalid
+        """
+        # Check inputs
+        if not isinstance(algorithm, CryptographicAlgorithm):
+            raise TypeError(
+                "algorithm must be a CryptographicAlgorithm enumeration")
+        elif not isinstance(length, six.integer_types) or length <= 0:
+            raise TypeError("length must be a positive integer")
+
+        # Verify that operations can be given at this time
+        if not self._is_open:
+            raise ClientConnectionNotOpen()
+
+        # Create the algorithm, length, and usage mask attributes
+        algorithm_attribute = self.attribute_factory.create_attribute(
+            AttributeType.CRYPTOGRAPHIC_ALGORITHM,
+            algorithm)
+        length_attribute = self.attribute_factory.create_attribute(
+            AttributeType.CRYPTOGRAPHIC_LENGTH,
+            length)
+        mask_attribute = self.attribute_factory.create_attribute(
+            AttributeType.CRYPTOGRAPHIC_USAGE_MASK,
+            [CryptographicUsageMask.ENCRYPT, CryptographicUsageMask.DECRYPT])
```

Comment:
Hmm, I think I'm fine either way. My original motivation here was to simple mimic what we've done with the prior tests, since we know those work with our backend(s). If you want to pull these, we can.